users/views.py

```
@@ -1,3 +1,40 @@
+        if form.is_valid():
+            user = form.get_user()
+            login(request, user)
+            messages.success(request, f"Logged in as {user.username}")
+            return redirect('chat_list')
+        else:
+            messages.error(request, "Invalid username or password.")
+    else:
+        form = AuthenticationForm()
+    return render(request, 'users/login.html', {'form': form})
+
+def logout(request):
```

Ruff (0.8.2)

Redefinition of unused `logout` from line 2

(F811)